locked-asset/factory/src/lib.rs

```diff
@@ -51,9 +55,29 @@ pub trait LockedAssetFactory:
         self.asset_token_id().set(&asset_token_id);
         self.default_unlock_period()
             .set(&UnlockPeriod { unlock_milestones });
+        self.set_extended_attributes_activation_nonce();
         Ok(())
     }
 
+    fn set_extended_attributes_activation_nonce(&self) {
+        //for extra safety
+        if self.extended_attributes_activation_nonce().is_empty() {
+            let one = BigUint::from(1u64);
+            let zero = BigUint::zero();
+            let mb_empty = ManagedBuffer::new();
+            let mv_empty = ManagedVec::new();
+            let token_id = self.locked_asset_token_id().get();
+
+            let nonce = self.send().esdt_nft_create(
```

Comment:
in order to pass all the tests - add another check that the contract has already the nft create role

Reply:
added a check for init_epoch not to be empty

Reply:
This `esdt_nft_create` is to get the latest nonce?

Reply:
yes